addons/binding/org.openhab.binding.kostalpikoiqplenticore/src/main/java/org/openhab/binding/kostalpikoiqplenticore/internal/things/KostalPlenticorePlus70WithoutBatteryHandler.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.binding.kostalpikoiqplenticore.internal.things;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.openhab.binding.kostalpikoiqplenticore.internal.KostalPikoIqPlenticoreHandlerBase;
+import org.openhab.binding.kostalpikoiqplenticore.internal.KostalPikoIqPlenticoreInverterTypes;
+
+/**
+ * The {@link KostalPlenticorePlus70WithoutBatteryHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author René Stakemeier - Initial contribution
+ */
+@NonNullByDefault
+public class KostalPlenticorePlus70WithoutBatteryHandler extends KostalPikoIqPlenticoreHandlerBase {
+
+    public KostalPlenticorePlus70WithoutBatteryHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    protected KostalPikoIqPlenticoreInverterTypes getInverterType() {
+        return KostalPikoIqPlenticoreInverterTypes.PlenticorePlus70WithoutBattery;
```

Comment:
It looks like all these handler classes are very similar. 

Why not just create a single base class which has a constructor that also accepts a `KostalPikoIqPlenticoreInverterTypes` parameter?

Then you can pass the proper values in the `KostalPikoIqPlenticoreHandlerFactory` and in the constructor assign it to a field.

That would allow for returning the appropriate value in this method.

Reply:
Optimized the code to not need the files :-)
This was a relict from my first steps, where I assumed, that for each thing defined in XML a corresponding java.class has to exist.